mazerunner/src/main/java/com/bugsnag/android/TestHarnessHooks.kt

```diff
@@ -2,12 +2,18 @@ package com.bugsnag.android
 
 import android.content.Context
 import android.net.ConnectivityManager
+import android.os.Handler
+import android.os.HandlerThread
 
 /**
  * Accesses the session tracker and flushes all stored sessions
  */
-internal fun flushAllSessions() {
-    Bugsnag.getClient().sessionTracker.flushStoredSessions()
+internal fun flushAllSessions(client: Client) {
+    val thread = HandlerThread("HandlerThread")
+    thread.start()
+    Handler(thread.looper).post {
+        client.sessionTracker.flushStoredSessions()
+    }
```

Comment:
Avoids performing network operations on the main thread, that could have caused scenarios to fail intermittently

Reply:
I don't entirely understand what this function is for; aren't sessions delivered automatically?